lightning/src/ln/peers/encryption.rs

```diff
@@ -0,0 +1,385 @@
+//! Handles all over the wire message encryption and decryption upon handshake completion.
+
+use ln::peers::{chacha, hkdf5869rfc};
+use util::byte_utils;
+use std::collections::VecDeque;
+
+pub(super) type SymmetricKey = [u8; 32];
+
+/// Maximum Lightning message data length according to
+/// [BOLT-8](https://github.com/lightningnetwork/lightning-rfc/blob/v1.0/08-transport.md#lightning-message-specification)
+/// and [BOLT-1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format):
+const LN_MAX_MSG_LEN: usize = ::std::u16::MAX as usize; // Must be equal to 65535
```

Comment:
So, I went back to try and understand where this came from and am pretty sure this is a bug.

https://github.com/lightningnetwork/lightning-rfc/blob/v1.0/08-transport.md#lightning-message-specification

The BOLT defines the maximum lightning message length as 65535, but the maximum encrypted size is 65569 due to the length and tags. This fails in some weird ways and am working a follow-up PR to fix the bug and make it cleaner.

Reply:
Done: `fix: Don't panic during valid decryption states`